backend/server/src/nix/jobs.rs

```diff
@@ -0,0 +1,46 @@
+use crate::nix::nix_eval_jobs::NixEvalItem;
+use std::io::{BufRead, BufReader};
+use std::path::Path;
+use std::process::{Command, Stdio};
+use tracing::debug;
+
+/// This file is meant to handle the evaluation of a "job" which is similar
+/// to the "jobset" by hydra, in particular:
+/// - You pass the file path of a nix file
+/// - You can optionally pass arguments to the file, which should be structured
+///   as a function which receives an attrset of inputs
+/// - The file outputs an [deeply nested] attrset of attrset<attr_path, drv>
+
+impl super::EvalService {
```

Comment:
I think it would be better to have type declarations and their impls in the same module. You could instead use this module to provide free standing function that implement the actual behavior and use the type to tie it all together.

Reply:
ah, so just receive a connection pool in this case? sounds fine to me

Reply:
done